### DISCUSSION TITLE
Can't assign cookies to a state

### DISCUSSION THREAD

**Discussion by falakjatin** [vercel/next.js#53070] [Help] [answered]
### Summary

I am trying new app route and I want to get cookies from client's browser and check if it is logged in. Due to the types I  am unable to assign the data from cookies to a state.

`const [userData, setUser] = useState<userDataType | null>(null)`

```
export interface userDataType {
    u_id: number,
    u_name: string,
    f_name: string,
    l_name: string,
}
```

```
const setUserData = (data: userDataType) => {
        cookies().set('data', JSON.stringify(data), { secure: true })
        setUser(data)
        router.push('/list')
    }
```

```
    const getData = () => {
        const data = cookies().get('data')
        if (data) {
            setUser(JSON.parse(data))
        }
    }
```

In above code line no 4 TypeScript is throwing an error "Argument of type 'RequestCookie' is not assignable to parameter of type 'string'."

<img width="1052" alt="image" src="https://github.com/vercel/next.js/assets/60683893/b4c5514c-b9e5-4955-8615-3317a79a721b">

If I remove the JSON.parse

<img width="939" alt="image" src="https://github.com/vercel/next.js/assets/60683893/c63e22df-7e8c-4be4-8143-366f9fce1843">

How to correctly type fix this?

**Comment by devjiwonchoi** [comment, marked as answer]
Modify `data` as `const data = cookies().get('data')?.value`.
Note that the type of `value` is `string | undefined`.
Since `JSON.parse` is inside the if condition of `data`, it'd be alright to not set a nullish coalescing for `data`.

```ts
const getData = () => {
    const data = cookies().get('data')?.value
    if (data) {
        setUser(JSON.parse(data))
    }
}
```
